registry/storage/driver/testsuites/testsuites.go

```diff
@@ -143,7 +158,10 @@ func (suite *DriverSuite) TestInvalidPaths(c *check.C) {
 
 	for _, filename := range invalidFiles {
 		err := suite.StorageDriver.PutContent(suite.ctx, filename, contents)
-		defer suite.StorageDriver.Delete(suite.ctx, firstPart(filename))
+		// only delete if file was succesfully written
```

Comment:
I don't think this is necessary.  If err == nil the test will fail the assert on line 165